specification/communication/Communication.Messages/models.tsp

```diff
@@ -205,6 +257,213 @@ model AudioNotificationContent extends NotificationContent {
+  @doc("Kind of MessageContent.")
+  kind: MessageContentKind;
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The message content of type text information.")
+model TextMessageContent extends MessageContent {
+  @doc("Message content kind is text.")
+  kind: MessageContentKind.text;
+
+  @doc("The text value.")
+  text: string;
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The message content of type document information.")
+model DocumentMessageContent extends MessageContent {
+  @doc("Message content kind is document.")
+  kind: MessageContentKind.document;
+
+  ...MediaMessageContent;
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The message content of type image information.")
+model ImageMessageContent extends MessageContent {
+  @doc("Message content kind is image.")
+  kind: MessageContentKind.image;
+
+  ...MediaMessageContent;
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The message content of type video information.")
+model VideoMessageContent extends MessageContent {
+  @doc("Message content kind is video.")
+  kind: MessageContentKind.video;
+
+  ...MediaMessageContent;
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The message content of type Button information.")
+model ButtonContent {
+  @doc("Unique Id of the button content.")
+  id: string;
+
+  @doc("Title of the button content.")
+  title: string;
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The message content of type ButtonSet/ List of buttons information.")
+model ButtonSetContent extends MessageContent {
+  @doc("Message content kind is Button.")
+  kind: MessageContentKind.buttonSet;
+
+  @doc("Unique Id of the button content.")
+  buttons: ButtonContent[];
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The message content of type Url information.")
+model LinkContent extends MessageContent {
+  @doc("Message content kind is url.")
+  kind: MessageContentKind.url;
+
+  @doc("Title of the url content.")
+  title: string;
+
+  @doc("The url in the content.")
+  url: url;
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The action content of type ActionGroup.")
+model ActionGroupContent extends MessageContent {
+  @doc("Message content kind is actionGroup.")
+  kind: MessageContentKind.group;
+
+  @doc("Title of the actionGroup content.")
+  title: string;
+
+  @doc("Set or group of actions.")
+  groups: ActionGroup[];
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The Action Group content.")
+model ActionGroup {
+  @doc("Title of the ActionGroup.")
+  title: string;
+
+  @doc("Array of items in ActionGroup.")
+  items: ActionGroupItem[];
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The Action group item in the content.")
+model ActionGroupItem {
+  @doc("Id of the Item.")
+  id: string;
+
+  @doc("Title of the Item.")
+  title: string;
+
+  @doc("Description of the Item.")
+  description: string;
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The media content.")
+model MediaMessageContent {
+  @doc("MediaUri of the media content.")
```

Comment:
```suggestion
  @clientName("mediaUrl", "java")
  @doc("MediaUri of the media content.")
```

Reply:
This is based on SDK review comment and same way is done for `MediaNotificationContentBase`

Reply:
Done , Please check the latest commit.